### ISSUE TITLE
implement non-llvm-backend release modes

### ISSUE THREAD

**Issue by andrewrk** [ziglang/zig#6035]
Right now self-hosted's design is tightly coupled to incremental linking and debug builds. It will take some effort to add support for the release mode use case.

#6032 will most likely provide the most common code path for release builds, however it will still be desirable to make release builds without the LLVM backend enabled. One simple reason would be architectures that Zig supports that LLVM does not.

**Comment by EleanorNB** [comment]
I don't think we should be intertwining an optional dependency so deeply with our toolchain. I think this should be implemented as a separate option, and have three configurable release modes rather than six monolithic ones. #5992.
